tools/online_editor/src/outline_widget.ts

```diff
@@ -82,12 +114,38 @@ function set_data(
 
     if (d.children != null) {
       const children_parent = document.createElement("ul");
-      set_data(d.children, children_parent, uri, goto_position);
+      current_pos_data.children = set_data(
+        d.children,
+        children_parent,
+        uri,
+        goto_position,
+      );
       row.appendChild(children_parent);
     }
 
+    pos_data.push(current_pos_data);
     parent.appendChild(row);
   }
+
+  return pos_data;
+}
+
+function deactivate_elements_and_find_to_activate(
```

Comment:
is this called at each character pressed?
(I'm a bit concerned about perfs, but maybe it's nothing)